### PR TITLE
Add big endian fixes for TD

### PR DESCRIPTION
**Pull request by giulianobelinassi**
Mainly ported from Arczi's Amiga port and the attempted Nintendo 64
port.

### CODE REVIEW

**Review by arczi84 — APPROVED** [review]
Looks fine

**Review by arczi84 — COMMENTED** [review]
Like above.

File: tiberiandawn/defines.h [hunk 1]
```diff
@@ -692,7 +708,11 @@ typedef enum HousesType : signed char
 #define HOUSEF_MULTI5  (1 << HOUSE_MULTI5)
 #define HOUSEF_MULTI6  (1 << HOUSE_MULTI6)
 
+#ifdef __BIG_ENDIAN__
 typedef enum PlayerColorType : signed char
+#else
+typedef enum PlayerColorType : signed int
+#endif
```
Comment (ccawley2011) [inline tiberiandawn/defines.h]: This seems to be different from the other changes in this file. Was that intentional?

File: common/crc.cpp [hunk 2]
```diff
@@ -111,6 +111,7 @@ int32_t CRCEngine::operator()(void const* buffer, size_t length)
         while (longcount--) {
             int32_t l;
             memcpy(&l, longptr++, sizeof(int32_t));
+            l = le32toh(l);
```
Comment (ccawley2011) [inline common/crc.cpp:114]: Would it be worth having a macro that encapsulates these three lines?

File: tiberiandawn/defines.h (hunk 1 above)
Comment (arczi84) [reply tiberiandawn/defines.h]: Yes that's wrong. Should be revesed.

Comment (giulianobelinassi) [reply tiberiandawn/defines.h]: @arczi84 How is this line on your amiga branch?

Comment (arczi84) [reply tiberiandawn/defines.h]: It's int
PlayerColorType : signed int

Comment (arczi84) [reply tiberiandawn/defines.h]: typedef enum PlayerColorType : signed char
is original the code

File: common/crc.cpp (hunk 2 above)
Comment (arczi84) [reply common/crc.cpp:114]: You can do whatever you like to make this cleaner as long it doesn't break anything.

**Review by arczi84 — APPROVED** [review]

File: tiberiandawn/defines.h (hunk 1 above)
Comment (giulianobelinassi) [reply tiberiandawn/defines.h]: I see. This is indeed reversed.

File: common/wwstd.h [hunk 3]
```diff
@@ -101,8 +101,11 @@ template <class T> T MAX(T a, T b)
 #define HIGH_WORD(a) ((unsigned int)(a) >> 16)
 
 // Merges to shorts to become a long
+#ifdef __BIG_ENDIAN__
+#define MAKE_LONG(a, b) ((long)((b)&0x0000FFFFL)) | ((long)(a) << 16)
```
Comment (OmniBlade) [inline common/wwstd.h]: Lets keep it the same as the updated code below and make this int despite the macro referring to long as long isn't 32bit on all platforms.

File: tiberiandawn/defines.h (hunk 1 above)
Comment (OmniBlade) [reply tiberiandawn/defines.h]: Why are all these extended to int for big endian? Issues with aligned access?

**Review by OmniBlade — CHANGES_REQUESTED** [review]
Overall I'd like to get this merged as it opens up portability to several older consoles to support big endian, just a few clarifications needed.

File: common/wwstd.h (hunk 3 above)
Comment (arczi84) [reply common/wwstd.h]: Yes, int should also work on BE.

File: tiberiandawn/defines.h (hunk 1 above)
Comment (arczi84) [reply tiberiandawn/defines.h]: Chars are swapped on big endian, I don't know a better solution for this.

**Comment by OmniBlade** [comment]
Closing as this PR is superceded by work done by @isojalka

**Pull request closed without merging**